devicetypes/plaidsystems/spruce-valve.src/spruce-valve.groovy

```diff
@@ -0,0 +1,76 @@
+/**
+ *  Copyright Plaid Systems 2020
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *
+
+11-2020
+ * valveDuration slider capability added back to presentation
+ * tabs and trim whitespace
+
+**/
+
+metadata {
+	definition (name: "Spruce Valve", namespace: "plaidsystems", author: "Plaid Systems", mnmn: "SmartThingsCommunity", vid: "558e73d3-c800-3669-b276-1d4352eda12b"){
+		capability "Actuator"
+		capability "Valve"
+		capability "Sensor"
+		capability "Health Check"
+		capability "heartreturn55003.valveDuration"
+
+		command "open"
+		command "close"
+		command "setValveDuration"
+
+		attribute "valveDuration", "NUMBER"
+	}
+}
+
+def installed() {
+	initialize()
+}
+
+def updated() {
+	initialize()
+}
+
+private initialize() {
+	//log.trace "Executing 'initialize'"
+
+	sendEvent(name: "valve", value: "closed")
+	sendEvent(name: "valveDuration", value: 5, unit: "mins")
+
+	sendEvent(name: "DeviceWatch-DeviceStatus", value: "online")
+	sendEvent(name: "healthStatus", value: "online")
+	sendEvent(name: "DeviceWatch-Enroll", value: [protocol: "cloud", scheme:"untracked"].encodeAsJson(), displayed: false)
```

Comment:
see comments about this on parent DTH

Reply:
Bump. Defining `checkInterval` should be enough, since parent's `ping()` is taking care of polling the device.

Reply:
Does child need any device health since parent takes care of this?  Can I remove capability "Health Check" and associated code.  And if I need to keep it, do I use checkInterval or the DeviceWatch-Enroll which sets it to cloud/untracked?

Reply:
I don't thin You need DeviceWatch-Enroll / untracked here. 
Try to leave "Health Check" here and redirect ping() calls to partent DTH -> 

def ping() {
	refresh()
}

def refresh() {
	// You will have to implement this method in spruce-controller
	parent.refreshChild()
       // or just call parent.refresh() if it already polls all the endpoints
 }

Reply:
additionally: https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/47339#discussion_r518332971